lib/bytes_by_prefix.c

```diff
@@ -62,12 +62,16 @@
 #define B_MULT	BSIZE		/* block size */
 #elif sgi
 #define B_MULT	BBSIZE		/* block size */
-#elif defined(__linux__) || defined(__sun) || defined(__hpux)
+#elif (defined(__linux__) || defined(__sun) || defined(__hpux)) && defined(DEV_BSIZE)
 #define B_MULT	DEV_BSIZE	/* block size */
```

Comment:
I agree it's a lot cleaner, but I'd rather not reduce support for platforms where there's a significant amount of code still in for them, I could work on removing it all in a separate patch series if you like?